TCC.Lib/Helpers/TaskHelper.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace TCC.Lib.Helpers
+{
+    public enum Fail
+    {
+        /// <summary>
+        /// Don't fail loop on exception, return a fault task
+        /// </summary>
+        Default,
+        /// <summary>
+        /// Fail loop as soon as an exception happends, return a sucessfull task, with exceptions in ParallelizedSummary
+        /// </summary>
+        Fast,
+        /// <summary>
+        /// Don't fail loop on exception, return a sucessfull task, with exceptions in ParallelizedSummary
+        /// </summary>
+        Smart
+    }
+    public class ParallelizedSummary
+    {
+        public IEnumerable<Exception> Exceptions { get; }
+
+        public bool IsCancelled { get; }
+
+        public bool IsSucess => !IsCancelled && (Exceptions == null || !Exceptions.Any());
+
+        public ParallelizedSummary(IEnumerable<Exception> exceptions, bool isCancelled)
+        {
+            Exceptions = exceptions;
+            IsCancelled = isCancelled;
+        }
+    }
+
+
+    public static class TaskHelper
+    {
+        private class ParallelizeCore : IDisposable
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Fix this implementation of 'IDisposable' to conform to the dispose pattern. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3881)